### DISCUSSION TITLE
A new kid on the block: Bun

### DISCUSSION THREAD

**Discussion by z3dev** [jscad/OpenJSCAD.org#1330] [General]
This looks very interesting, and takes JavaScript performance to another level. 

https://bun.sh/

Has anyone looked at this?

**Comment by hrgdavor** [comment]
@z3dev I am following bun progress closely.

One issue is that windows build is not ready yet (WSL is not a viable compromise for me aside from testing bun a bit)
Another issue related to my recent work is that there is specific problem when trying to test webcomponent in bun native test lib.

Window support is inching towards completion, but was anounced and postponed few times, hence this cool website https://isbunwindowsyet.com

Bun is using safari js runtime instead of V8, and there is not a huge performance difference in running apps, so It could maybe run jscad faster but I am not expecting huge gains.

Where bun shines is buidling. Their move to add most used and most useful parts of building process into runtime does give huge performance boost.
 - testing api
 - http server
 - bundler that transpiles and runs js,ts,jsx,tsx but does not do checking like `tsc` so speed is close to esbuild
 - shell api is also very promising to help with many of the projects automation tasks

having these core things builtin, would make some of my projects node_modules 99% smaller if using bun.


On the other hand there is a huge perf gain we can get for jscad, and I have some testing samples ready to explore what is needed and how it works... https://github.com/jscad/OpenJSCAD.org/discussions/822#discussioncomment-8891463
